src/transformers/modeling_flax_bert.py

```diff
@@ -0,0 +1,366 @@
+
+        # Layer Norm
+        layer_norm = BertLayerNorm(name="layer_norm")(summed_emb)
+
+        return layer_norm
+
+
+class BertAttention(nn.Module):
+    num_heads: int
+    head_size: int
+
+    @compact
+    def __call__(self, hidden_state, attention_mask):
+        self_att = nn.attention.SelfAttention(
+            num_heads=self.num_heads, qkv_features=self.head_size, name="self")(hidden_state, attention_mask)
+
+        layer_norm = BertLayerNorm(name="layer_norm")(self_att + hidden_state)
+        return layer_norm
+
+
+class BertIntermediate(nn.Module):
+    output_size: int
+
+    @compact
+    def __call__(self, hidden_state):
+        # TODO: Add ACT2FN reference to change activation function
+        dense = nn.Dense(features=self.output_size, name="dense")(hidden_state)
+        return gelu(dense)
+
+
+class BertOutput(nn.Module):
+    @compact
+    def __call__(self, intermediate_output, attention_output):
+        h = nn.Dense(attention_output.shape[-1], name="dense")(intermediate_output)
+        h = BertLayerNorm(name="layer_norm")(h + attention_output)
+        return h
+
+
+class BertLayer(nn.Module):
+    num_heads: int
+    head_size: int
+    intermediate_size: int
+
+    @compact
+    def __call__(self, hidden_state, attention_mask):
+        attention = BertAttention(self.num_heads, self.head_size, name="attention")(hidden_state, attention_mask)
+        intermediate = BertIntermediate(self.intermediate_size, name="intermediate")(attention)
+        output = BertOutput(name="output")(intermediate, attention)
+
+        return output
+
+
+class BertLayerCollection(nn.Module):
+    """
+    Stores N BertLayer(s)
+    """
+    num_layers: int
+    num_heads: int
+    head_size: int
+    intermediate_size: int
+
+    @compact
+    def __call__(self, inputs, attention_mask):
+        assert self.num_layers > 0, "num_layers should be >= 1, got ({})".format(self.num_layers)
+
+        # Initialize input / output
+        input_i = output_i = inputs
+
+        # Forward over all encoders
+        for i in range(self.num_layers):
+            layer = BertLayer(self.num_heads, self.head_size, self.intermediate_size, name="{}".format(i))
+            output_i = layer(input_i, attention_mask)
+            input_i = output_i
+        return output_i
+
+
+class BertEncoder(nn.Module):
+    num_layers: int
+    num_heads: int
+    head_size: int
+    intermediate_size: int
+
+    @compact
+    def __call__(self, hidden_state, attention_mask):
+        layer = BertLayerCollection(self.num_layers, self.num_heads, self.head_size, 
+            self.intermediate_size, name="layer")(hidden_state, attention_mask)
+        return layer
+
+
+class BertPooler(nn.Module):
+    @compact
+    def __call__(self, hidden_state):
+        first_token = hidden_state[:, 0]
+        out = nn.Dense(hidden_state.shape[-1], name="dense")(first_token)
+        return jax.lax.tanh(out)
+
+
+class BertModel(nn.Module):
```

Comment:
As you guessed, I choose to follow what is already widely adopted in the library regarding modules fragmentation. 

This is something our users are welcoming because they like how easy it is to tweak one module. Also it makes almost a 1-1 matchs with both PyTorch & TensorFlow implementations, so it might be easier for users who would like to give it a try to do the move.

Does it make sense?